include/seqan3/range/detail/inherited_iterator_base.hpp

```diff
@@ -312,19 +341,29 @@ class inherited_iterator_base : public std::conditional_t<std::is_pointer_v<base
     }
 
     //!\brief Return underlying container value currently pointed at.
-    constexpr decltype(auto) operator[](std::make_signed_t<difference_type> const n) const
+    constexpr decltype(auto) operator[](std::make_signed_t<difference_type> const n)
```

Comment:
why is the difference type unsigned? That would not be valid per concept defintion as it has to be signed_integer_like

Reply:
> why is the difference type unsigned?

Is it?

Reply:
yes

Reply:
I don't know, but why is it wrapped in make_signed?

Reply:
It resolves to the underlying difference_type. Apparently we inherited something somewhere that was unsigned. I don't recall entirely. It's not new code so I would pref to leave it like this. I will revisit this file when we have move-only iterators after the next range upgrade.

Reply:
ok